security/security-design/fn-datasafe-dbaudit-to-oci-logging/terraform/function/oci-datasafe-audit-to-logging/func.py

```diff
@@ -0,0 +1,416 @@
+###############################################################################
+# Copyright (c) 2022, 2023, Oracle and/or its affiliates.  All rights reserved.
+# This software is dual-licensed to you under the Universal Permissive License
+# (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl.
+###############################################################################
+#
+# Author: Fabrizio Zarri
+#
+################################################################################
+
+import json
+import oci
+import io
+import logging
+import pandas as pd
+
+from oci.util import to_dict
+from fdk import response
+from datetime import datetime, timedelta
+
+
+
+
+def init_datasafe_client():
+    try:
+        rps = oci.auth.signers.get_resource_principals_signer()
+        datasafe_client = oci.data_safe.DataSafeClient({}, signer=rps)
+    except Exception as e:
+        logging.getLogger().error("Init DataSafe client Error: %s", e)
```

Comment:
Don't call logging.getLogger() more than once in your code. Do it after the import and reuse it.
Example:

logger = logging.getLogger()
...
logger.error("Error message")

This applies to all your python code!